### PR TITLE
London Class 7- Ali-Nosratipour-Javascript- core1- coursework-Week3

### PR DESCRIPTION
**Pull request by alinosratipour**
# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

### CODE REVIEW

File: 1-exercises/A-array-find/exercise.js [hunk 1]
```diff
@@ -15,10 +15,15 @@ var names = [
   "Anastasia",
   "Karim",
   "Ahmed",
+  
 ];
 
-var longNameThatStartsWithA = findLongNameThatStartsWithA(names);
 
+function findLongNameThatStartsWithA(name){
+  return  name.startsWith("A") && name.length > 7
```
Comment (Laura-TM) [inline 1-exercises/A-array-find/exercise.js:23]: Love the use of .startsWith(); I had not seen that one or remembered

File: 1-exercises/C-array-every/exercise.js [hunk 2]
```diff
@@ -5,7 +5,11 @@
 var students = ["Omar", "Austine", "Dany", "Swathi", "Lesley", "Rukmini"];
 var group = ["Austine", "Dany", "Swathi", "Daniel"];
 
-var groupIsOnlyStudents; // complete this statement
+function checkStudent(n){
```
Comment (Laura-TM) [inline 1-exercises/C-array-every/exercise.js:8]: Perhaps 'n' is a little vague and it could be labelled a bit clearer

**Comment by Laura-TM** [comment]
Overall, really resourceful and effective, well done; just thought of something that could make it easier for other developers while reading your code.

File: 2-mandatory/7-password-validator.js [hunk 3]
```diff
@@ -21,16 +21,44 @@ Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
 
 */
+/** 
+
+*/
+
+
+
+const check = [/[a-z]/ , /[A-Z]/, /[0-9]/, /[!#$%,&*]/];
```
Comment (ademalagoz) [inline 2-mandatory/7-password-validator.js:30]: Really nice shot Ali. Love it,
Well done

File: 1-exercises/D-array-filter/exercise.js [hunk 4]
```diff
@@ -8,7 +8,13 @@
 
 var pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1], false, "whoops"];
 
-var pairsByIndex; // Complete this statement
+function CheckMembers(e){
+   return Array.isArray(e) ? e.length === 2  : false;
```
Comment (he-man-david) [inline 1-exercises/D-array-filter/exercise.js:12]: great job

File: 1-exercises/F-array-forEach/exercise.js [hunk 5]
```diff
@@ -9,6 +9,23 @@
 
 var arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15];
 
+function findResult(num) {
+	if (num % 3 === 0 && num % 5 === 0){
```
Comment (he-man-david) [inline 1-exercises/F-array-forEach/exercise.js:13]: nice job putting this at top

File: 2-mandatory/1-oxygen-levels.js [hunk 6]
```diff
@@ -11,7 +11,16 @@
     Some string methods that might help you here are .replace() and .substring(). 
 */
 
-function safeLevels() {}
+
+function safeLevels(arr) {   
+    //Convert our Array item from string to float Integer
+    const ConvertToNumber = arr.map(item => {
+       return parseFloat(item);
```
Comment (he-man-david) [inline 2-mandatory/1-oxygen-levels.js:18]: nice use of parseFloat

File: 2-mandatory/5-journey-planner.js [hunk 7]
```diff
@@ -20,8 +20,8 @@
 function checkCodeIsThere(stringText) {
   let magicWord = "code";
   //edit code below
-  if (stringText) {
-    return stringText;
+  if (stringText.includes(magicWord)) {
```
Comment (he-man-david) [inline 2-mandatory/5-journey-planner.js:23]: do we need includes() here?

File: 2-mandatory/5-journey-planner.js [hunk 8]
```diff
@@ -123,6 +129,10 @@ function getLocationName() {}
 */
 function journeyPlanner(locations, transportMode) {
   // Implement the function body
+   const Planner = locations.filter( el => el.includes(transportMode))
+   .map (el => el[0])
```
Comment (he-man-david) [inline 2-mandatory/5-journey-planner.js:133]: do we need the .map() here?

File: 2-mandatory/7-password-validator.js [hunk 9]
```diff
@@ -21,16 +21,44 @@ Expected Result:
...
 
 function validatePasswords(passwords) {
+const validPassword = passwords.map((item,index,arr)=>{
+       
+  if(item.length >= 5 &&
```
Comment (he-man-david) [inline 2-mandatory/7-password-validator.js:35]: great job! shows that you really understand

**Pull request closed without merging**